homeassistant/components/acer_projector/switch.py

```diff
@@ -89,24 +86,46 @@ def _write_read(self, msg):
         try:
             if not self.ser.is_open:
                 self.ser.open()
+            else:
+                self.ser.flushInput()
+                self.ser.flushOutput()
+
             msg = msg.encode('utf-8')
             self.ser.write(msg)
-            # Size is an experience value there is no real limit.
-            # AFAIK there is no limit and no end character so we will usually
-            # need to wait for timeout
-            ret = self.ser.read_until(size=20).decode('utf-8')
+            # Try to get a response a maximum of 5 times
+            if self.ser.inWaiting() > 0:
+                _LOGGER.debug('Characters to read %s', self.ser.inWaiting())
+                # Changed to read number of waiting characters
+                test_list = self.ser.read(self.ser.inWaiting()).split(b'\r')
```

Comment:
I/O and data parsing should be moved to a standalone library published on pypi.

Reply:
Sorry @MartinHjelmare but this is a joke. Should have been mentioned when I made the original changes. The original code didn't have this and this is an improvement so why are we making it so hard to fix broken things?

Reply:
Yes, it should preferably have been mentioned in the first review round. I'm sorry we didn't catch this then.

We shouldn't add more protocol into home assistant since that should be in a separate library. If a fix only removes things it would be ok. Adding protocol related things isn't ok.

Reply:
Spent far too much of my own time on this to date @MartinHjelmare. Feel free to close this PR and others can live with a broken component. Will keep mine as a custom.